x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformAction.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.action;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.TaskOperationFailure;
+import org.elasticsearch.action.support.tasks.BaseTasksRequest;
+import org.elasticsearch.action.support.tasks.BaseTasksResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class TriggerTransformAction extends ActionType<TriggerTransformAction.Response> {
+
+    public static final TriggerTransformAction INSTANCE = new TriggerTransformAction();
+    public static final String NAME = "cluster:admin/transform/trigger";
+
+    private TriggerTransformAction() {
+        super(NAME, TriggerTransformAction.Response::new);
+    }
+
+    public static class Request extends BaseTasksRequest<Request> {
+
+        private final String id;
+
+        public Request(String id, TimeValue timeout) {
+            this.id = id;
```

Comment:
`ExceptionsHelper.requireNonNull(id, TransformField.ID.getPreferredName());` maybe?

Reply:
timeout must be non-null it seems?